pylint/checkers/variables.py

```diff
@@ -819,6 +822,38 @@ def _uncertain_nodes_in_false_tests(
 
         return uncertain_nodes
 
+    @staticmethod
+    def _node_guarded_by_same_test(node: nodes.NodeNG, other_if: nodes.If) -> bool:
+        """Identify if `node` is guarded by an equivalent test as `other_if`.
```

Comment:
There's a lot of inference here, but the primer time vary in the same range than it usually does, so the performance impact is probably acceptable.

Reply:
Yeah, I'm assuming that all of these nodes are inferred anyway elsewhere in the variables checker, so we're hitting the cache.